local_planner/src/nodes/local_planner.cpp

```diff
@@ -126,6 +125,9 @@ void LocalPlanner::create2DObstacleRepresentation(const bool send_to_fcu) {
   }
   polar_histogram_ = new_histogram;
 
+  // create 3D points from combined histogram
+  reprojectPoints(polar_histogram_);
```

Comment:
That PR could save lots time in buildLookAheadTree().
However,since reprojectPoints histogram in the current stage,  the function reprojectPoints() should not use last position "position_old_", the reprojected histogram is based on current position.
https://github.com/PX4/avoidance/blob/4a92ae0a6fbfd3d654657808ecfa8c494149bcfa/local_planner/src/nodes/local_planner.cpp#L361-L364.

Reply:
good catch! I missed that one :)